### PR TITLE
Plugin babysteps.

### PR DESCRIPTION
**Pull request by pq**
@bwilkerson @keertip

### CODE REVIEW

**Comment by keertip** [comment]
lgtm!

File: lib/src/plugin/linter_plugin.dart [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  @override
+  void registerExtensionPoints(RegisterExtensionPoint registerExtensionPoint) {
+    lintRuleExtensionPoint = registerExtensionPoint(
+        LINT_RULE_EXTENSION_POINT, _validateTaskExtension);
+  }
+
+  @override
+  void registerExtensions(RegisterExtension registerExtension) {
+    _rules.forEach((LintRule rule) =>
```
Comment (bwilkerson) [inline lib/src/plugin/linter_plugin.dart]: This is fine, but if you wanted to you could get rid of the '_rules' field by in-lining it here. As it is, you have effectively stored the collection in two places.

**Comment by bwilkerson** [comment]
LGTM

File: lib/src/plugin/linter_plugin.dart (hunk 1 above)
Comment (pq) [inline lib/src/plugin/linter_plugin.dart]: Good point.  I've done away with `_rules` in favor of inlining.

Thanks!

**Pull request merged**